Sources/OktaOAuth2/Extensions/URL+Extensions.swift

```diff
@@ -0,0 +1,99 @@
+//
+// Copyright (c) 2023-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+// The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+//
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and limitations under the License.
+//
+
+import Foundation
+import AuthFoundation
+
+extension URL {
+    typealias RedirectError = AuthorizationCodeFlow.RedirectError
+    
+    func oauth2QueryComponents(redirectUri: URL?, state: String?) throws -> [String: String] {
+        guard let components = URLComponents(url: self,
+                                             resolvingAgainstBaseURL: false)
+        else {
+            throw RedirectError.invalidRedirectUrl
+        }
+        
+        guard components.scheme?.lowercased() == redirectUri?.scheme?.lowercased()
+        else {
+            throw RedirectError.unexpectedScheme(components.scheme)
+        }
+        
+        guard var query = components.queryItems?.reduce(into: [String: String](), { partialResult, queryItem in
+            if let value = queryItem.value {
+                partialResult[queryItem.name] = value
+            }
+        }) else {
+            throw RedirectError.missingQueryArguments
+        }
+        
+        if let state = state {
+            guard query["state"] == state else {
+                throw RedirectError.invalidState(query["state"])
+            }
+        }
```

Comment:
We do not have a query param of state being appended to our redirect url. Because of that, this state check and throw prevents the error description from getting set.

Reply:
our redirect looks like com.company.site://auth?error=mappingnotfound&error_description=No%20mappings.

Reply:
@shanepadgett Yes, the state is returned in some circumstances, which happens to be the cases I was testing with. I have an updated commit that removes the `state` value check from the error handling.

Reply:
Great. Will test again

Reply:
That worked!